core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/CryptoUtil.java

```diff
@@ -306,10 +329,19 @@ public String encryptAndBase64Encode(byte[] plainText) throws
                 decryptedValue = encryption.decryptWithSymmetricKey(cipherTextBytes);
             } else {
                 Cipher keyStoreCipher;
-                KeyStoreManager keyMan = KeyStoreManager
-                        .getInstance(MultitenantConstants.SUPER_TENANT_ID, this.getServerConfigService(), this.getRegistryService());
-                KeyStore keyStore = keyMan.getPrimaryKeyStore();
-                PrivateKey privateKey = (PrivateKey) keyStore.getKey(keyAlias, keyPass.toCharArray());
+                KeyStore keyStore;
+                PrivateKey privateKey;
+                KeyStoreManager keyMan = KeyStoreManager.getInstance(
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        this.getServerConfigService(),
+                        this.getRegistryService());
+                if (keyMan.getInternalKeyStore() != null) {
```

Comment:
This code segment seems to be duplicated. Will it be useful to convert this to a separate method?